doc/csv/recipes.rdoc

```diff
@@ -152,6 +156,51 @@ Output:
   #<CSV::Row "Name":"bar" "Value":"1">
   #<CSV::Row "Name":"baz" "Value":"2">
 
+=== Parsing: Field Converters
+
+==== Convert Fields to Objects
+
+Use field converters to change parsed Strings into other, more specific, object.
+
+==== Convert Fields to Objects Using Built-In Converters
+
+Without converters (all fields parsed as Strings):
+  source = "0,1.1,2020-09-19"
+  parsed = CSV.parse(source)
+  parsed # => [["0", "1.1", "2020-09-19"]]
+  parsed.first.each {|field| p field.class }
+Output:
+  String
+  String
+  String
+
+With built-in converters (see {Built-In Field Converters}[../../CSV.html#class-CSV-label-Built-In+Field+Converters]):
+  parsed = CSV.parse(source, converters: :all)
+  parsed # => [[0, 1.1, #<DateTime: 2020-09-19T00:00:00+00:00 ((2459112j,0s,0n),+0s,2299161j)>]]
+  parsed.first.each {|field| p field.class }
+Output:
+  Integer
+  Float
+  DateTime
+
+==== Convert Fields to Objects Using Custom Converters
+
+See {Custom Field Converters}[../../CSV.html#class-CSV-label-Custom+Field+Converters].
+
+Define a custom field converter and add it to the \Converters \Hash:
+  strip_converter = proc {|field| field.strip }
+  CSV::Converters[:strip] = strip_converter
+
+Without the new converter:
+  string = " foo , 0 \n bar , 1 \n baz , 2 \n"
+  array = CSV.parse(string)
+  array # => [[" foo ", " 0 "], [" bar ", " 1 "], [" baz ", " 2 "]]
+
+With the new converter:
+  string = " foo , 0 \n bar , 1 \n baz , 2 \n"
+  array = CSV.parse(string, converters: strip_converter)
```

Comment:
We don't need to register to `CSV::Converters` if we specify the custom converter proc to `converters:`:

```ruby
require "csv"

strip_converter = proc {|field| field.strip }
string = " foo , 0 \n bar , 1 \n baz , 2 \n"
array = CSV.parse(string, converters: strip_converter)
p array
# [["foo", "0"], ["bar", "1"], ["baz", "2"]]

Reply:
If we register to `CSV::Converters`, we can specify the registered key instead of the proc:

```ruby
require "csv"

strip_converter = proc {|field| field.strip }
CSV::Converters[:strip] = strip_converter
string = " foo , 0 \n bar , 1 \n baz , 2 \n"
array = CSV.parse(string, converters: :strip)
p array
# [["foo", "0"], ["bar", "1"], ["baz", "2"]]
```

Reply:
Fixed (as unregistered) in next push.

Converter registration needs clarification over in lib/csv.rb.  That's for another PR.

Reply:
Added snippet to show registration.  Link to lib/csv.rb no longer needed.